lib/directory-monitor-poll.c

```diff
@@ -123,15 +157,32 @@ _start_watches(DirectoryMonitor *s)
 {
   DirectoryMonitorPoll *self = (DirectoryMonitorPoll *)s;
   GDir *directory = NULL;
+  GError *error = NULL;
+  gchar *fullpath = NULL;
+  GStatBuf stat_result;
+
   directory = g_dir_open(self->super.real_path, 0, NULL);
   if (directory)
     {
       const gchar *filename = g_dir_read_name(directory);
       while (filename)
         {
-          collection_comparator_add_initial_value(self->comparator, filename);
+          fullpath = build_filename(self->super.real_path, filename);
+          if (g_stat(fullpath, &stat_result))
+            {
+              msg_debug("Cannot stat file/directory",
+                        evt_tag_str("filename", fullpath),
+                        evt_tag_str("error", error->message));
```

Comment:
The `error` is not used before - but initialized to `NULL` - thus `error->message` is a null pointer dereference.

Reply:
Now fixed for the ``stat`` block. FYI - ``error`` is actually used:
```
  GError *error = NULL;
  gchar *fullpath = NULL;
  GStatBuf stat_result;
  GDir *directory = g_dir_open(self->super.real_path, 0, &error);
```

Reply:
@Kokan - now ``error`` usage removed from all occurrences relating to this PR functionality. I left the one relating to ``open_dir`` since it's not related to this PR.